### DISCUSSION TITLE
No import option for identity brokering

### DISCUSSION THREAD

**Discussion by tremaze** [keycloak/keycloak#10372] [Archive]
### Description

Add option to not import users to the local database when acting as a broker. When all relevant information is available in remote database, it's useless to store the data again in a "cache" like local database.

Old issue: https://issues.redhat.com/browse/KEYCLOAK-4429

**Comment by gmeazza** [comment]
This is an important issue. Storing personal data for no reason is not just useless, it causes major problems with privacy regulations. Is there any hope this will be addressed anytime soon?

**Comment by stianst** [comment]
Not anytime soon. It's something we have in mind for the new store though, but won't be considered until after other priorities are completed.

**Reply by jamalkm** [reply]
Is there any update on this?

**Comment by shivendushekhar** [comment]
Hi,

I have few questions: -

- What is the purpose of creating this local database? Is it performance or scale?
- What level of user data is stored locally? Complete or partial? Is there a way to limit the data which is getting stored?
- How is the data getting stored? In plain text or encrypted? Is it possible to control the format of data getting stored?
- Are password hashes also getting stored? 

It seems to an important fix, particularly from the point of view of data ownership and underlying security concerns. Is there any plan to fix it in coming releases?

Thanks.

**Comment by igueye** [comment]
Keycloak 23 release introduced `transient-users` for identity brokering as an experimental feature. 
When using Keycloak as an identity broker, this experimental feature allows an admin to configure the Keycloak server to not create users in its local database when a user authenticates with an external identity provider.
You may need to refer to the following resources:
* https://github.com/keycloak/keycloak/issues/11334 / https://github.com/keycloak/keycloak/pull/23977
* https://github.com/keycloak/keycloak/issues/24278 / https://github.com/keycloak/keycloak/pull/24333
* https://github.com/keycloak/keycloak/discussions/26637 (where feedback are welcomed)
